tests/units/Glpi/Toolbox/VersionParser.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units\Glpi\Toolbox;
+
+/**
+ * Test class for src/Glpi/Toolbox/versionparser.class.php
+ */
+class VersionParser extends \GLPITestCase {
+
+   protected function versionsProvider() {
+      return [
+         [
+            'version'             => '',
+            'keep_stability_flag' => false,
+            'normalized'          => '',
+            'stable'              => true,
+            'dev'                 => false,
+         ],
+         [
+            'version'             => '9.5+2.0',
+            'keep_stability_flag' => false,
+            'normalized'          => '9.5+2.0', // not semver compatible, cannot be normalized
+            'stable'              => true,
+            'dev'                 => false,
+         ],
+         [
+            'version'             => '0.89',
+            'keep_stability_flag' => false,
+            'normalized'          => '0.89.0',
+            'stable'              => true,
+            'dev'                 => false,
+         ],
+         [
+            'version'             => '9.2',
+            'keep_stability_flag' => false,
+            'normalized'          => '9.2.0',
+            'stable'              => true,
+            'dev'                 => false,
+         ],
+         [
+            'version'             => '9.2',
+            'keep_stability_flag' => true, // should have no effect
+            'normalized'          => '9.2.0',
+            'stable'              => true,
+            'dev'                 => false,
+         ],
+         [
+            'version'             => '9.4.1.1',
+            'keep_stability_flag' => false,
+            'normalized'          => '9.4.1',
```

Comment:
Not sure here. `9.4.1.1` is not the same release than `9.4.1`, but normalized version will tell it is... Maybe should we consider that this is not a semver compatible version?

Reply:
Normalization is used by update process. Whether you have the 9.4.1 or the 9.4.1.1 version, you will have to run the 9.4.1->9.4.2 migration.
We could adapt this part, but I am not sure it will be usefull for us.

Reply:
OK for me then